server/tso/tso.go

```diff
@@ -71,7 +74,7 @@ func (t *timestampOracle) setTSOPhysical(next time.Time) {
 	t.tsoMux.Lock()
 	defer t.tsoMux.Unlock()
 	// make sure the ts won't fall back
-	if typeutil.SubTimeByWallClock(next, t.tsoMux.physical) >= updateTimestampGuard {
+	if typeutil.SubTimeByWallClock(next, t.tsoMux.physical).Milliseconds() >= 0 {
```

Comment:
How about changing the implementation of `SubTimeByWallClock`?

Reply:
I think we should keep `SubTimeByWallClock` to use nanosecond precision and leave it up to the caller to decide exactly what precision to use. Although we need to use millisecond precision for TSO comparisons, we still need nanoseconds for physical system time comparison in the `UpdateTimestamp` and `SyncTimestamp` functions.